src/main/resources/static/js/testrules.js

```diff
@@ -177,91 +178,105 @@ jQuery(document).ready(
+        validateJSONAndUpload(subscriptionFile, false);
+        $(this).val("");
+      };
+
       //Upload events list json data
       $(".container").on("click", "button.upload_rules", function(event) {
         event.stopPropagation();
         event.preventDefault();
+        var isRules = true;
+        replaceAppendModal(isRules);
 
-        function createRulesUploadWindow() {
-          if (document.createEvent) {
-            var event = document.createEvent('MouseEvents');
-            event.initEvent('click', true, true);
-            pomRules.dispatchEvent(event);
-          } else {
-            pomRules.click();
-          }
-        }
-        
-        function createUploadWindowMSExplorer() {
-          $('#upload_rules').click();
-          var file = $('#upload_rules').prop('files')[0];
-          validateJSONAndUpload(file, true);
-        }
-
-        // HTML5 Download File window handling
-        createRulesUploadWindow();
       });
 
-      //Upload list of events json data
+       //Upload list of events json data
       $(".container").on("click", "button.upload_events", function(event) {
         event.stopPropagation();
         event.preventDefault();
+        var isRules = false;
+        replaceAppendModal(isRules);
+      });
 
-        function createUploadWindow() {
-          if (document.createEvent) {
-            var event = document.createEvent('MouseEvents');
-            event.initEvent('click', true, true);
-            pomEvents.dispatchEvent(event);
-          } else {
-            pomEvents.click();
-          }
-        }
+      function replaceAppendModal(isRules){
+        $('#AppendReplaceModal').modal('show');
 
-        function createUploadWindowMSExplorer() {
-          $('#upload_events').click();
-          var file = $('#upload_events').prop('files')[0];
-          validateJSONAndUpload(file, false);
-        }
+        document.getElementById('replaceButton').onclick = function(){
+            $('#AppendReplaceModal').modal('hide');
+            isReplacing = true;
+            if(isRules){
+                createRulesUploadWindow();
+            }else{
+                createUploadWindow();
+            }
+        };
 
+        document.getElementById('appendButton').onclick = function(){
+            $('#AppendReplaceModal').modal('hide');
+            isReplacing = false;
+            if(isRules){
+                createRulesUploadWindow();
+            }else{
+                createUploadWindow();
+            }
+        };
+      }
 
-        // HTML5 Download File window handling
-        createUploadWindow();
-      });
+     function createRulesUploadWindow() {
+        if (document.createEvent) {
+          var event = document.createEvent('MouseEvents');
+          event.initEvent('click', true, true);
+          pomRules.dispatchEvent(event);
+        } else {
+          pomRules.click();
+        }
+      }
+
+      function createUploadWindow() {
```

Comment:
I'm not sure I understand why this function is different to the one above, createRulesUploadWindow? As far as I can see they contain the exact same code?

Reply:
createRulesUploadWindow change pomRules and createUploadWIndow change pomEvents

Reply:
Got it! Missed that part apparently